### PR TITLE
feat: open port

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: N/A

### Overview

Add an [open port to k8s service](https://juju.is/docs/sdk/open-a-kubernetes-port-in-your-charm#heading--add-a-kubernetes-service-port-to-your-charm) which is a good information for SREs when deploying the charm.

### Rationale

Helps annotate the k8s service with port definitions.

### Juju Events Changes

The pebble ready event sets open port when charm is successfully initialized.

### Module Changes

`charm.py`

### Library Changes

None.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -147,18 +147,37 @@ def test__on_jenkins_pebble_ready_get_version_error(
...
+):
+    """
+    arrange: given a mocked jenkins get_version raises TimeoutError on the second call.
+    act: when the Jenkins pebble ready event is fired.
+    assert: Jenkins falls into BlockedStatus.
+    """
+    harness = harness_container.harness
+    # monkeypatch environment variables because the test is running in self-hosted runners and juju
+    # proxy environment is picked up, making the test fail.
+    monkeypatch.setattr(state.os, "environ", {})
```
[the remark this answers is not in the document]
Comment (yanksyoon) [reply tests/unit/test_charm.py]: Good point! Fixed thank you!

File: tests/unit/conftest.py [hunk 2]
```diff
@@ -429,3 +429,11 @@ def mock_charm_fixture():
     mock_charm = unittest.mock.MagicMock(spec=CharmBase)
     mock_charm.app.planned_units.return_value = 1
     return mock_charm
+
+
+@pytest.fixture(scope="function", name="patch_os_environ")
```
Comment (jdkandersson) [inline tests/unit/conftest.py:434]: What about just passing the `autouse` argument so it doesn't have to be manually applied to every test? Or is it not needed on some tests?

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -147,32 +146,43 @@ def test__on_jenkins_pebble_ready_get_version_error(
...
-    harness_container: HarnessWithContainer,
-    monkeypatch: pytest.MonkeyPatch,
-    exception: Exception | None,
-    expected_status: ops.StatusBase,
+    harness_container: HarnessWithContainer, monkeypatch: pytest.MonkeyPatch
 ):
     """
     arrange: given a mocked jenkins client and a patched requests instance.
     act: when the Jenkins pebble ready event is fired.
     assert: the unit status should show expected status.
```
Comment (cbartz) [inline tests/unit/test_charm.py:182]:
```suggestion
    assert: the unit status should show expected status and the jenkins port should be open
```

**Review by cbartz — DISMISSED** [review]
LGTM, just a small comment

File: tests/unit/test_charm.py (hunk 3 above)
Comment (yanksyoon) [reply tests/unit/test_charm.py]: Updated. Thank you!

**Review by jdkandersson — APPROVED** [review]

**Review by cbartz — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]

**Review by jdkandersson — APPROVED** [review]

**Review by cbartz — APPROVED** [review]

**Pull request merged**